File: app/src/main/java/com/ivantha/ihaveatheory/model/Scenario.java
package com.ivantha.ihaveatheory.model;

import com.ivantha.ihaveatheory.game.scene.Scene;
import com.ivantha.ihaveatheory.game.scene.location.LocationScene;
import com.ivantha.ihaveatheory.game.scene.start.ForestScene;
import com.ivantha.ihaveatheory.game.scene.start.StartScene;

import java.io.Serializable;

public class Scenario implements Serializable{
    private int dayBackground;
    private int nightBackground;

    private Scene scene;

    public void setForestScenario(){
        ForestScene forestScene = new ForestScene();
        setScene(forestScene);
    }

    public int getDayBackground() {
        return dayBackground;
    }

    public void setDayBackground(int dayBackground) {
        this.dayBackground = dayBackground;
    }

    public int getNightBackground() {
        return nightBackground;
    }

    public void setNightBackground(int nightBackground) {
        this.nightBackground = nightBackground;
    }

    public Scene getScene() {
        return scene;
    }

    public void setScene(Scene scene) {
        this.scene = scene;

        if (scene instanceof StartScene){
            dayBackground = ((StartScene)scene).getDayBackground();
            nightBackground = ((StartScene)scene).getNightBackground();
        }
        else if(scene instanceof LocationScene){
            dayBackground = ((LocationScene)scene).getDayBackground();
            nightBackground = ((LocationScene)scene).getNightBackground();
        }
    }
}

File: app/src/main/java/com/avoid/ihaveatheory/model/Scenario.kt
package com.avoid.ihaveatheory.model

import com.avoid.ihaveatheory.game.scene.Scene
import com.avoid.ihaveatheory.game.scene.location.LocationScene
import com.avoid.ihaveatheory.game.scene.start.ForestScene
import com.avoid.ihaveatheory.game.scene.start.StartScene

import java.io.Serializable

class Scenario : Serializable {
    var dayBackground: Int = 0
    var nightBackground: Int = 0

    var scene: Scene? = null
        set(scene) {
            field = scene

            if (scene is StartScene) {
                dayBackground = scene.dayBackground
                nightBackground = scene.nightBackground
            } else if (scene is LocationScene) {
                dayBackground = scene.dayBackground
                nightBackground = scene.nightBackground
            }
        }

    fun setForestScenario() {
        val forestScene = ForestScene()
        scene = forestScene
    }
}
